apps/shell/app.js

```diff
@@ -55,9 +57,44 @@ app.use(process.env.PASSENGER_BASE_URI || '/', router);
 var server = new http.createServer(app);
 var wss = new WebSocket.Server({ noServer: true });
 
+whitelist = [];
+if (process.env.SSHHOST_WHITELIST){
+  whitelist = process.env.SSHHOST_WHITELIST.split(':');
+}
+
+glob.sync('/etc/ood/config/clusters.d/*.y*ml').forEach(function (file){
+  var data = yaml.safeLoad(fs.readFileSync(file));
+  const {
+    v2: {
+      metadata: {
+        hidden = true
+      },
+      login
+    }
+  } = data
+
+  if (login){ //check login separately, doesn't exist in some config files
+    const{
+      host = 'default',
+      default: isDefault = false
+    } = login
+
+    if (!hidden){
+      if(isDefault){
+        whitelist.unshift(host);
+      } else {
+        whitelist.push(host);
+      }
+    }
+  }
+});
+
+whitelist = Array.from(new Set(whitelist)); // remove duplicates
+default_sshhost = whitelist[0]; //default sshhost if not set
+
 wss.on('connection', function connection (ws, req) {
   var match;
-  var host = process.env.DEFAULT_SSHHOST || 'localhost';
+  var host = process.env.DEFAULT_SSHHOST || default_sshhost || 'owens.osc.edu'; //worst case scenario, w/no config files or env variable, use owens
```

Comment:
We don't want to use OSC defaults here because other sites install this software too.

```suggestion
  const host = process.env.DEFAULT_SSHHOST || default_sshhost
```